api/base/serializers.py

```diff
@@ -29,6 +29,59 @@ def _url_val(val, obj, serializer, **kwargs):
         return val
 
 
+class HyperlinkedIdentityFieldWithMeta(ser.HyperlinkedIdentityField):
+    # Returns a list with a url and an optional hash containing additional meta information
+
+    def __init__(self, view_name=None, **kwargs):
+        kwargs['read_only'] = True
+        kwargs['source'] = '*'
+        self.meta = kwargs.pop('meta', None)
+        self.link_type = kwargs.pop('link_type', 'url')
+        super(ser.HyperlinkedIdentityField, self).__init__(view_name, **kwargs)
+
+    def get_url(self, obj, view_name, request, format):
+        """
+        Given an object, return the URL that hyperlinks to the object.
+
+        Returns null if lookup value is None
+        """
+
+        if getattr(obj, self.lookup_field) is None:
+            return None
+
+        return super(ser.HyperlinkedIdentityField, self).get_url(obj, view_name, request, format)
+
+    def to_representation(self, value):
+        request = self.context.get('request', None)
+        format = self.context.get('format', None)
```

Comment:
Again, a lot of this looks like it was copied from the parent class's implementation. This should raise a red flag.

Consider the [Open Close Principle](http://www.oodesign.com/open-close-principle.html). Always see if you extend--rather than overwrite--a class's behavior.

If you can't avoid overriding, then try to re-use the parent classes' behavior.

```python
def to_representation(self, value):
    url = super(HyperlinkedIdentityFieldWithMeta, self).to_representation(
        value
    )
    ret = {'url': url, 'link_type': self.link_type}
    if self.meta:
        ret['meta'] = self.meta
    return ret
```